nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/PutAzureBlobStorage_v12.java

```diff
@@ -121,19 +168,42 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
             if (createContainer && !containerClient.exists()) {
                 containerClient.create();
             }
+
             BlobClient blobClient = containerClient.getBlobClient(blobName);
+            Relationship relationship = REL_SUCCESS;
+
+            final BlobRequestConditions blobRequestConditions = new BlobRequestConditions();
 
-            long length = flowFile.getSize();
+            try {
+                if (conflictStrategy == AzureStorageConflictStrategy.OVERWRITE_IF_SOURCE_NEWER) {
+                    OffsetDateTime ifUnmodifiedSince = OffsetDateTime.ofInstant(
+                            Instant.ofEpochMilli(flowFile.getEntryDate()),
```

Comment:
While I did experiment with a pair of property to control the format and attribute name in case that resolution strategy was set to `OVERWRITE_IF_SOURCE_NEWER` – I ultimately decided that there is probably not a strong enough use-case to warrant the added complexity.

And it's easy to bring this functionality in if the need should arise.